src/Bicep.LangServer/Handlers/GetDeploymentDataHandler.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Bicep.Core.Emit;
+using Bicep.Core.Workspaces;
+using Bicep.LanguageServer.CompilationManager;
+using MediatR;
+using Microsoft.Extensions.Logging;
+using OmniSharp.Extensions.JsonRpc;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Bicep.LanguageServer.Handlers
+{
+    [Method("bicep/getDeploymentData", Direction.ClientToServer)]
+    public record GetDeploymentDataRequest(TextDocumentIdentifier TextDocument)
+        : ITextDocumentIdentifierParams, IRequest<GetDeploymentDataResponse?>;
+
+    public record GetDeploymentDataResponse(string TemplateJson, string? ParametersJson);
+
+    public class GetDeploymentDataHandler : IJsonRpcRequestHandler<GetDeploymentDataRequest, GetDeploymentDataResponse?>
+    {
+        private readonly ILogger<BicepDocumentSymbolHandler> logger;
+        private readonly ICompilationManager compilationManager;
+
+        public GetDeploymentDataHandler(ILogger<BicepDocumentSymbolHandler> logger, ICompilationManager compilationManager)
+        {
+            this.logger = logger;
+            this.compilationManager = compilationManager;
+        }
+
+        public async Task<GetDeploymentDataResponse?> Handle(GetDeploymentDataRequest request, CancellationToken cancellationToken)
+        {
+            await Task.Yield();
```

Comment:
>Yield [](http://example.com/codeflow?start=23&length=5)

What's the reason for adding `Yield()` here?

Reply:
Pure laziness - wrapping everything in `Task.FromResult` is a hassle.